integrations/lambda/demo.ipynb

```diff
@@ -0,0 +1,1451 @@
+    "                        raise Exception(f\"Measure template was missing name: {measure_template}\")\n",
+    "                    measure_name = measure_template[\"name\"]\n",
+    "\n",
+    "                    # Optional template fields\n",
+    "                    measure_value_type = \"DOUBLE\"\n",
+    "                    if \"type\" in measure_template:\n",
+    "                        measure_value_type = str(measure_template[\"type\"]).strip().upper()\n",
+    "                    max_variation = 1.5\n",
+    "                    if \"max_variation\" in measure_template:\n",
+    "                        max_variation = measure_template[\"max_variation\"]\n",
+    "                    max_value = 100.0\n",
+    "                    if \"max\" in measure_template:\n",
+    "                        max_value = measure_template[\"max\"]\n",
+    "                    min_value = 0.0\n",
+    "                    if \"min\" in measure_template:\n",
+    "                        min_value = measure_template[\"min\"]\n",
+    "                    varchar_length = 10\n",
+    "                    if \"varchar_length\" in measure_template:\n",
+    "                        varchar_length = measure_template[\"varchar_length\"]\n",
+    "\n",
+    "                    measure = {\n",
+    "                        \"MeasureName\": measure_name,\n",
+    "                        \"MeasureValueType\": measure_value_type\n",
+    "                    }\n",
+    "\n",
+    "                    measure_value = None\n",
+    "                    if \"random_options\" in measure_template:\n",
+    "                        measure_value = random.choice(measure_template[\"random_options\"])\n",
+    "                    else:\n",
+    "                        if current_date == start_date:\n",
+    "                            if measure_value_type == \"DOUBLE\":\n",
+    "                                measure_value = random.uniform(min_value, max_value)\n",
+    "                            elif measure_value_type == \"VARCHAR\":\n",
+    "                                measure_value = self._generate_random_string(varchar_length)\n",
+    "                            elif measure_value_type == \"BIGINT\":\n",
+    "                                measure_value = int(random.uniform(min_value, max_value))\n",
+    "                            elif measure_value_type == \"BOOLEAN\":\n",
+    "                                measure_value = random.choice(True, False)\n",
+    "                            else:\n",
+    "                                raise Exception(\"Measure value type not recognized\")\n",
+    "                        else:\n",
+    "                            if measure_value_type == \"DOUBLE\":\n",
+    "                                measure_value = max(min_value, min(entity[\"latest_measures\"][measure_name] + random.uniform(-max_variation, max_variation), max_value))\n",
```

Comment:
This is the method for iterating on previous measure values for an entity, creating a gradually changing measure value over time.

Since the list of records is flat and finding the latest measure values for an entity would require a scan, the entity dict is re-used to store the latest measure. This could be replaced with the index of the measure value in the list instead, if storage is a concern.